controllers/texttemplate_controller_test.go

```diff
@@ -0,0 +1,381 @@
+package controllers
+
+import (
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"math/rand/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+	"time"
+
+	templatesv1alpha1 "github.com/kluctl/template-controller/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("TextTemplate controller", func() {
+	const (
+		timeout  = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	Context("Template with no inputs and simple string template", func() {
+		ns := fmt.Sprintf("test-%d", rand.Int64())
```

Comment:
Ginkgo discourages this style of variable initialization, ref. https://onsi.github.io/ginkgo/#avoid-spec-pollution-dont-initialize-variables-in-container-nodes.

Reply:
I'll think about this more. Right now, the tests each build on the side-effects of the previous ones (I assume that's another anti-pattern...but it makes things so much easier) and thus the suggested `BeforeEach` style would not work for now.